common/src/services/crypto.service.ts

```diff
@@ -131,6 +131,24 @@ export class CryptoService implements CryptoServiceAbstraction {
         return keyHash == null ? null : this.keyHash;
     }
 
+    async compareAndUpdateKeyHash(masterPassword: string, key: SymmetricCryptoKey): Promise<boolean> {
+        const storedKeyHash = await this.getKeyHash();
+        if (masterPassword != null && storedKeyHash != null) {
+            const localKeyHash = await this.hashPassword(masterPassword, key, HashPurpose.LocalAuthorization);
+            if (localKeyHash != null && storedKeyHash === localKeyHash) {
+                return true;
+            }
+
+            const serverKeyHash = await this.hashPassword(masterPassword, key, HashPurpose.ServerAuthorization);
```

Comment:
May want to put a TODO comment here that we can remove this at some point in the future?